### PR TITLE
TT-10675 add SQS Pump Backend support

### PR DESCRIPTION
**Pull request by masoudhaghbin**
Add SQS pump to supported backends
## Description

The PR will add the support to pump logs into a SQS instance

## Related Issue

https://github.com/TykTechnologies/tyk-pump/issues/737

## Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [x] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [x] I have updated the documentation accordingly.
- [x] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

File: pumps/sqs.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+		return SQSError
+	}
+	s.log.Debug("ElapsedTime in seconds for ", len(data), " records:", time.Now().Sub(startTime))
+	s.log.Info("Purged ", len(data), " records...")
+	return nil
+}
+
+func (s *SQSPump) write(c context.Context, messages []types.SendMessageBatchRequestEntry) error {
+	pretty.Print(messages)
```
Comment (mativm02) [inline pumps/sqs.go:139]: I think this could be a `log.Debug`

File: pumps/sqs.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	EnvPrefix                        string `mapstructure:"meta_env_prefix"`
+	QueueName                        string `mapstructure:"aws_queue_name"`
+	AWSRegion                        string `mapstructure:"aws_region"`
+	AWSSecret                        string `mapstructure:"aws_secret"`
+	AWSKey                           string `mapstructure:"aws_key"`
+	AWSEndpoint                      string `mapstructure:"aws_endpoint"`
+	AWSDelaySeconds                  int32  `mapstructure:"aws_delay_seconds"`
+	AWSMessageGroupID                string `mapstructure:"aws_message_group_id"`
+	AWSSQSBatchLimit                 int    `mapstructure:"aws_sqs_batch_limit"`
+	AWSMessageIDDeduplicationEnabled bool   `mapstructure:"aws_message_id_deduplication_enabled"`
```
Comment (mativm02) [inline pumps/sqs.go:51]: Could you add an explanatory comment to each field?

**Review by mativm02 — CHANGES_REQUESTED** [review]
Hey @masoudhaghbin! Thank you for your contribution, we really appreciate it!
Overall, it looks pretty good. I just left 2 small suggestions. 

The only missing points are:
* Add this new pump to the readme (you can follow [this example of another pump](https://github.com/TykTechnologies/tyk-pump/blob/master/README.md#resurfaceio))
* Add unit tests for this new pump (you can follow [this unit test file example of another pump](https://github.com/TykTechnologies/tyk-pump/blob/master/pumps/splunk_test.go))

We're looking forward to merging this PR once the suggested changes are addressed. Thank you for your patience!

File: pumps/sqs.go (hunk 2 above)
Comment (masoudhaghbin) [reply pumps/sqs.go]: @mativm02 Hi
Thanks for your review and notes you mentioned. 
I've committed new changes according to them.

**Review by mativm02 — APPROVED** [review]
lgtm!

**Comment by mativm02** [comment]
@masoudhaghbin I created a duplicated version of this PR with all your commits so CIs can effectively run without complaining about some repo secrets. I'll close this one so we can keep working on the new one and merge it. You can take a look at it [here](https://github.com/TykTechnologies/tyk-pump/pull/774). Again, thank you for your contribution!

**Pull request closed without merging**